delta-api/src/main/java/io/cdap/delta/api/Offset.java

```diff
@@ -50,6 +50,13 @@ public boolean equals(Object o) {
     Offset offset = (Offset) o;
     return Objects.equals(state, offset.state);
   }
+  @Override
+  public String toString() {
+    if (state.isEmpty()) {
+      return null;
```

Comment:
We should not return null from toString as it can lead to unexpected NPE. Instead we can return string "state=null"

Reply:
I will remove this check. In case of null, it will return "{}" (empty map) while logging for offset.